### PR TITLE
Fe2 CI got stuck

### LINKED ISSUE #1608: [BUG] CI tests of FE2 sometimes get stuck
### Description (Actual behavior)
When executing the tests on the CI it happens sometimes that the fe2 tests get stuck.

### Expected behavior
It shouldn't happen.

### Version & Environment
I've gone through the github action history to see when this behaviour first appeared, and it was during the PR #1575, so the problem could be in those modifications .

##### Front-ends:
- [ ] Fe1-Web
- [X] Fe2-Android
- [ ] Not applicable

##### Back-ends:
- [ ] Be1-Go
- [ ] Be2-Scala
- [X] Not applicable

### Workaround
Usually canceling the workflow and restarting the test fixes this.

### Impact
In terms of deployment this could slow a bit the process.

### Possible root cause
Looking at the logs, the lock happens generally when testing the repositories, so my best guess is that there is a deadlock, since the tests are executed in parallel. Also in the repos we used synchronized methods, so this can be totally plausible

### PR DESCRIPTION
**Pull request by matteosz**
This PR resolves #1608

### CODE REVIEW

**Review by pierluca — DISMISSED** [review]
This looks deceptively simple. I'm happy to approve, but I don't know if I'm missing some subtlety here 😅

**Comment by pierluca** [comment]
The CI (still) seems broken :-/

**Review by K1li4nL — APPROVED** [review]
For what its worth: LGTM, I never developped on android but what I am reading make sense, at least mocking these elements should hopefully make it easier to debug if the problem  comes up again

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/repository/WitnessingRepositoryTest.java [hunk 1]
```diff
@@ -67,11 +87,20 @@ public class WitnessingRepositoryTest {
...
           new ArrayList<>());
 
+  @Rule(order = 0)
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
   @Before
   public void setUp() {
     Application application = ApplicationProvider.getApplicationContext();
-    appDatabase =
-        AppDatabaseModuleHelper.getAppDatabase(ApplicationProvider.getApplicationContext());
```
Comment (simone-kalbermatter) [inline fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/repository/WitnessingRepositoryTest.java:74]: I don't really understand this line, was the problem that it was not using a mock?

Comment (matteosz) [reply fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/repository/WitnessingRepositoryTest.java:74]: The Room database instance created was shared among several thread instances running the tests in parallel, thus creating concurrency issues

**Pull request merged**